scripts/test/Muon/elemental_analysis/elemental_analysis_test.py

```diff
@@ -696,6 +696,42 @@ def test_line_delayed_checked_checks_line_and_calls_add_line_by_type(self, mock_
         self.gui.lines.delayed.setChecked.assert_called_with(True)
         self.assertEqual(1, mock_add_line_by_type.call_count)
 
+    @mock.patch('Muon.GUI.ElementalAnalysis.elemental_analysis.ElementalAnalysisGui._remove_element_lines')
+    def test_deselect_elements(self,mock_remove_element_lines):
+
+        self.gui.ptable.deselect_element = mock.Mock()
+
+        self.gui.peaks.enable_deselect_elements_btn = mock.Mock()
+        self.gui.peaks.disable_deselect_elements_btn = mock.Mock()
+
+        self.gui.deselect_elements()
+
+        self.assertEquals(self.gui.peaks.enable_deselect_elements_btn.call_count , 1)
+        self.assertEquals(self.gui.peaks.disable_deselect_elements_btn.call_count , 1)
+
+        self.assertEquals(self.gui.ptable.deselect_element.call_count , len(self.gui.element_widgets))
+        self.assertEquals(mock_remove_element_lines.call_count , len(self.gui.element_widgets))
+
+        calls = [mock.call(element) for element in self.gui.element_widgets.keys()]
+        self.gui.ptable.deselect_element.assert_has_calls(calls)
+        mock_remove_element_lines.assert_has_calls(calls)
+
+    @mock.patch('Muon.GUI.ElementalAnalysis.elemental_analysis.ElementalAnalysisGui._remove_element_lines')
+    def test_deselect_elements_fails(self,mock_remove_element_lines):
+
+        self.gui.ptable.deselect_element = mock.Mock()
+
+        self.gui.peaks.enable_deselect_elements_btn = mock.Mock()
+        self.gui.peaks.disable_deselect_elements_btn = mock.Mock()
+
+        self.gui.deselect_elements()
+
+        with self.assertRaises(AssertionError):
```

Comment:
The with is there so if code in the with block does not raise an AssertionError one will will be raised and if code in block raises an AssertionError then it will be caught by block. The png is there to show new gui as changes affected gui's view.

Reply:
Ahh okay, makes sense. I would maybe change the comment to be a bit more concise, something like "test passes when deselect_element is not called with hydrogen".